### PR TITLE
Allow viewing collectible images on full screen

### PR DESCRIPTION
**Pull request by grilix**
Allows collectible images to be shown on full screen, with the ability to zoom it in.

image

image

image

### CODE REVIEW

File: src/screens/Collectible/Collectible.js [hunk 1]
```diff
@@ -111,6 +115,12 @@ class CollectibleScreen extends React.Component<Props> {
     this.props.navigation.navigate(SEND_COLLECTIBLE_FROM_ASSET_FLOW, { assetData });
   };
 
+  onTouchImage = () => {
+    const { navigation: { state: { params: { assetData } } } } = this.props;
```
Comment (jsidorenko) [inline src/screens/Collectible/Collectible.js]:
```suggestion
    const { navigation: { navigate, state: { params: { assetData } } } } = this.props;
```

File: src/screens/Collectible/Collectible.js [hunk 2]
```diff
@@ -111,6 +115,12 @@ class CollectibleScreen extends React.Component<Props> {
...
+
+    this.props.navigation.navigate(FULL_SCREEN_COLLECTIBLE, { collectible: assetData });
```
Comment (jsidorenko) [inline src/screens/Collectible/Collectible.js]:
```suggestion
    navigate(FULL_SCREEN_COLLECTIBLE, { collectible: assetData });
```

File: src/screens/Collectible/FullScreenCollectible.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+const genericCollectible = require('assets/images/no_logo.png');
+
+class FullScreenCollectible extends React.PureComponent<Props> {
+  collectible: Collectible;
+
+  constructor(props: Props) {
+    super(props);
+
+    const { navigation: { state: { params: { collectible } } } } = props;
```
Comment (jsidorenko) [inline src/screens/Collectible/FullScreenCollectible.js]:
```suggestion
    const collectible = props.navigation.getParam('collectible')
```

File: src/screens/Collectible/FullScreenCollectible.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  justify-content: center;
+  min-height: 300px;
+  min-width: 300px;
+  align-self: center;
+`;
+
+const genericCollectible = require('assets/images/no_logo.png');
+
+class FullScreenCollectible extends React.PureComponent<Props> {
+  collectible: Collectible;
```
Comment (jsidorenko) [inline src/screens/Collectible/FullScreenCollectible.js]: Just wondering is it allowed to use internal variables for PureComponents?

File: src/screens/Collectible/Collectible.js (hunk 2 above)
Comment (grilix) [reply src/screens/Collectible/Collectible.js]: Damn, they need to add multi-line suggestions :sweat_smile:

File: src/screens/Collectible/FullScreenCollectible.js (hunk 4 above)
Comment (grilix) [reply src/screens/Collectible/FullScreenCollectible.js]: It looks like it is :smile:! Good catch, didn't realize I did this.

**Comment by grilix** [comment]
@jsidorenko thanks for the comments; it turns out we don't even need a new screen since `react-native-image-view` shows the image on a modal, which makes everything simpler and gives the user a better experience.

Please let me know your thoughts.

File: src/screens/Collectible/Collectible.js [hunk 5]
```diff
@@ -97,20 +103,58 @@ const CollectibleImage = styled(CachedImage)`
...
+  };
+
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
     const isFocused = this.props.navigation.isFocused();
     if (!isFocused) {
       return false;
     }
-    const isEq = isEqual(this.props, nextProps);
-    return !isEq;
+    if (!isEqual(this.state, nextState)) {
```
Comment (vyvil) [inline src/screens/Collectible/Collectible.js]: Those could be combined: 
```suggestion
    if (!isEqual(this.state, nextState) || !isEqual(this.props, nextProps)) return true;
```

File: src/screens/Collectible/Collectible.js [hunk 6]
```diff
@@ -97,20 +103,58 @@ const CollectibleImage = styled(CachedImage)`
...
+        title: name,
+      },
+    ];
+
+    return (
+      <ImageView
+        images={imageViewImages}
+        imageIndex={0}
+        isVisible={isImageViewVisible}
+        onClose={this.onCloseImageView}
```
Comment (vyvil) [inline src/screens/Collectible/Collectible.js]: Would be great to add close icon that we use on other modals. There's a way to add custom close component using controls prop: https://www.npmjs.com/package/react-native-image-view#controls-prop

File: src/screens/Collectible/Collectible.js (hunk 5 above)
Comment (grilix) [reply src/screens/Collectible/Collectible.js]: Updated

File: src/screens/Collectible/Collectible.js (hunk 6 above)
Comment (grilix) [reply src/screens/Collectible/Collectible.js]: Updated

**Pull request merged**